pkg/translator/azure/resourcelogs_to_logs.go

```diff
@@ -125,6 +126,18 @@ func (r ResourceLogsUnmarshaler) UnmarshalLogs(buf []byte) (plog.Logs, error) {
 	return l, nil
 }
 
+func getTimestamp(record azureLogRecord) (pcommon.Timestamp, error) {
+	var s string
+	if record.Time != "" {
+		s = record.Time
+
+	} else if record.Timestamp != "" {
+		s = record.Timestamp
+	}
+
+	return asTimestamp(s)
+}
+
```

Comment:
A few suggestions here:

First, regarding the way the result of this function is handled (currently skips the log if the timestamp cannot be parsed): The timestamp field is optional in our data model, so typically if there is an error parsing, you could (arguably _should_) continue creating the log. There is an [ObservedTimestamp field](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-observedtimestamp) which will automatically be populated and exporters may use this in place of "timestamp" if necessary.

Malformed timestamps aside, this function may intentionally pass an empty string to `asTimestamp` which will always generate an error. You might consider avoiding that call altogether and returning a static error immediately.

Finally, a very minor change is that you might consider saving an allocation for every log record by not declaring `var s string`:

```suggestion
func getTimestamp(record azureLogRecord) (pcommon.Timestamp, error) {
	if record.Time != "" {
		return asTimestamp(record.Time)
	}
	if record.Timestamp != "" {
		return asTimestamp(record.Timestamp)
	}
	return errNoTimestamp
}
```